engine/mpm_solver.py

```diff
@@ -43,7 +43,7 @@ class MPMSolver:
         'SEPARATE': surface_separate
     }
 
-    grid_size = 4096
+    grid_size = 6144
```

Comment:
Can we expose this attribute to the user and keep its original `4096` value? :-) Also you probably want a power of two here.

Reply:
https://github.com/taichi-dev/taichi_elements/blob/347381c5ebad71273486672f043598e153b046ac/engine/mpm_solver.py#L138-L143
Does it sound good to automatically increase `grid_size` when `res` is larger or equal to `grid_size//2`? This guarantees all sides of the simulation region to be unbounded.

Reply:
Sounds good :-)